server/src/main/java/org/candlepin/async/tasks/RefreshPoolsForProductJob.java

```diff
@@ -0,0 +1,147 @@
+import org.candlepin.service.SubscriptionServiceAdapter;
+import org.candlepin.util.Util;
+
+import com.google.inject.Inject;
+
+import org.apache.log4j.MDC;
+
+import java.util.Objects;
+
+
+/**
+ * RefreshPoolsForProductJob
+ */
+public class RefreshPoolsForProductJob implements AsyncJob {
+
+    private PoolManager poolManager;
+    private ProductCurator productCurator;
+    private SubscriptionServiceAdapter subAdapter;
+    private OwnerServiceAdapter ownerAdapter;
+
+    public static final String LAZY_REGEN = "lazy_regen";
+
+    public static final String JOB_KEY = "REFRESH_POOLS_FOR_PRODUCT";
+    private static final String JOB_NAME = "refresh_pools_for_product";
+
+    @Inject
+    public RefreshPoolsForProductJob(
+        final ProductCurator productCurator,
+        final PoolManager poolManager,
+        final SubscriptionServiceAdapter subAdapter,
+        final OwnerServiceAdapter ownerAdapter) {
+        this.poolManager = Objects.requireNonNull(poolManager);
+        this.productCurator = Objects.requireNonNull(productCurator);
+        this.subAdapter = Objects.requireNonNull(subAdapter);
+        this.ownerAdapter = Objects.requireNonNull(ownerAdapter);
+    }
+
+    @Override
+    public String execute(JobExecutionContext context) {
+        final JobArguments args = context.getJobArguments();
+
+        final String productUuid = args.getAsString(JobStatus.TARGET_ID);
+        final Boolean lazy = args.getAsBoolean(LAZY_REGEN);
+        final StringBuilder result = new StringBuilder();
+
+        final Product product = this.productCurator.get(productUuid);
+
+        if (product != null) {
+            final Refresher refresher = poolManager.getRefresher(this.subAdapter, this.ownerAdapter, lazy);
+
+            refresher.add(product);
+            refresher.run();
+
+            result.append("Pools refreshed for product: ")
+                .append(productUuid)
+                .append("\n");
+        }
+        else {
+            result.append("Unable to refresh pools for product \"")
+                .append(productUuid)
+                .append("\": Could not find a product with the specified UUID");
+        }
+
+        return result.toString();
+    }
+
+    /**
+     * Creates a JobConfig configured to execute the export job. Callers may further manipulate
+     * the JobConfig as necessary before queuing it.
+     *
+     * @return
+     *  a JobConfig instance configured to execute the export job
+     */
+    public static JobConfig createJobConfig(final Product product, final boolean lazy) {
+        return new RefreshPoolsForProductConfig(product, lazy);
+    }
+
+    /**
+     * Job configuration object for the export job
+     */
+    public static class RefreshPoolsForProductConfig extends JobConfig {
+
+        RefreshPoolsForProductConfig(final Product product, final boolean lazy) {
+            this.setJobKey(JOB_KEY)
+                .setJobName(JOB_NAME + Util.generateUUID())
```

Comment:
UUID generation is no longer needed for the new job system

Reply:
Done.